File: SecondYearCS/OOP/CivitasRB/lib/CasillaSorpresa.rb
require_relative 'sorpresa.rb'
require_relative 'casilla.rb'
require_relative 'tipo_sorpresa.rb'
require_relative 'titulo_propiedad.rb'
require_relative 'mazo_sorpresa.rb'
require_relative 'diario.rb'
require_relative 'dado.rb'
module CivitasRB
  class CasillaSorpresa < Casilla
    def initialize(mazo,nombre)
      super(nombre)
      @mazo = mazo
    end

    def recibeJugador(act,todos)
      if self.jugadorCorrecto(act, todos)
        self.informe(act,todos)
        sorpresa = @mazo.siguiente()
        sorpresa.aplicarAJugador(act,todos)
      end
    end

  end
end

File: SecondYearCS/OOP/Civitas/src/civitas/CasillaSorpresa.java
package civitas;
import java.util.ArrayList;

public class CasillaSorpresa extends Casilla {

    private MazoSorpresa mazo;              // tipo:: SORPRESA
    
    //Tipo Sorpresa
    CasillaSorpresa(MazoSorpresa mazo, String nombre){
        super(nombre);
        this.mazo = mazo;
    }

    @Override 
    void recibeJugador(int actual, ArrayList<Jugador> todos) throws Exception{
        if(this.jugadorCorrecto(actual,todos)){
            this.informe(actual,todos);
            Sorpresa sorpresa = this.mazo.siguiente();
            sorpresa.aplicarAJugador(actual, todos);
        }
    }

    @Override
    public String toString(){
       String result = super.toString();
       return result;
    }
}
